### PR TITLE
Update the guide for KerasTuner custom training loop

### PR DESCRIPTION
**Pull request by haifeng-jin**

### CODE REVIEW

File: guides/keras_tuner/custom_tuner.py [hunk 1]
```diff
@@ -161,53 +131,81 @@ def run_train_step(data):
...
+`MyHyperModel.fit()` to `tuner.search()`.
 
-mnist_train = tf.data.Dataset.from_tensor_slices((x_train, y_train))
+"""
+
+tuner.search(x=x_train, y=y_train, validation_data=(x_val, y_val))
 
-tuner.search(train_ds=mnist_train)
+"""
+Finally, we can retrieve the results.
```
Comment (fchollet) [inline guides/keras_tuner/custom_tuner.py:208]: At the end of the guide, please add a couple of sentences with key takeaways from the guide and pointers to other resources.

**Review by fchollet — COMMENTED** [review]
Thanks for the PR! I applied some copyedits. Looks good to me overall!

**Review by fchollet — APPROVED** [review]
Thank you!

**Comment by fchollet** [comment]
I'm not able to generate the guide, seems like we have an uncompiled model:

> RuntimeError: You must compile your model before training/testing. Use `model.compile(optimizer, loss)`.

Can you take a look?

**Comment by haifeng-jin** [comment]
I have generated files without error.
The error may be caused by your local KT version, it needs to be 1.1.0 or later.
https://github.com/keras-team/keras-io/pull/780

**Pull request merged**